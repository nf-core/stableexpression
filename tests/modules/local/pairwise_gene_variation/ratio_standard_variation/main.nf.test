nextflow_process {

    name "Test Process RATIO_STANDARD_VARIATION"
    script "modules/local/pairwise_gene_variation/ratio_standard_variation/main.nf"
    process "RATIO_STANDARD_VARIATION"
    tag "ratio_std"
    tag "module"

    test("Should run without failures") {

        when {
            process {
                """
                file = file( '$projectDir/tests/test_data/pairwise_gene_variation/expression_ratio/output/ratios.0.1.parquet', checkIfExists: true)
                input[0] = file
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
