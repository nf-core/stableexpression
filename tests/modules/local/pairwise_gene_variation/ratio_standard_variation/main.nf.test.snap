{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "std.0.1.parquet:md5,7a1529e9be1f9ca8aecb9146c39ab4f5"
                ],
                "1": [
                    [
                        "RATIO_STANDARD_VARIATION",
                        "python",
                        "3.12.8"
                    ]
                ],
                "2": [
                    [
                        "RATIO_STANDARD_VARIATION",
                        "polars",
                        "1.17.1"
                    ]
                ],
                "data": [
                    "std.0.1.parquet:md5,7a1529e9be1f9ca8aecb9146c39ab4f5"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-29T16:12:29.850020618"
    }
}